### PR TITLE
[MRG] backport github actions to stable branch (3.5.x)

### PR DESCRIPTION
**Pull request by ctb**
This backports GitHub Actions and associated test stuff from #1244 #1249 #1252 #696 #1295 over to the stable branch, mostly to support releases along 3.5.x.

## Checklist

- [ ] Is it mergeable?
- [ ] `make test` Did it pass the tests?
- [ ] `make coverage` Is the new code covered?
- [ ] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [ ] Was a spellchecker run on the source code and documentation after
  changes were made?

### CODE REVIEW

**Comment by ctb** [comment]
this appears to be working? but now I would need to re-enable python 2.7...

**Comment by ctb** [comment]
ok, now looks like tox needs to be configured to actually run tests :)

**Comment by ctb** [comment]
...how weird, this is running on an incorrect git ref? https://github.com/dib-lab/sourmash/runs/1893907938?check_suite_focus=true

**Comment by ctb** [comment]
@luizirber I think this is ready for review and merge, as the final cherry atop the whipped cream of sourmash 3.5.1. We're still having problems with pysam install on py3.8 and py3.9, but fixing them is annoying and the tests do pass on py3.7. Most importantly, the wheels build, and the wheels upload, and pip install works in a clean py3.8.6 environment (`conda create -n sourmash-install-test-2 python=3.8`).

FWIW, if we were to pursue pysam fixes, I would probably look into using Cython to rebuild the C files, and if that didn't work, I might want to try running inside of conda (which can install pysam just fine).

File: setup.py [hunk 1]
```diff
@@ -80,7 +80,7 @@ def build_native(spec):
         'doc' : ['sphinx', 'myst-parser[sphinx]', 'alabaster',
                  "sphinxcontrib-napoleon", "nbsphinx",
                  "ipython"],
-        '10x': ['bam2fasta==1.0.4'],
+        '10x': ['bam2fasta==1.0.8'],
```
Comment (luizirber) [inline setup.py]: does the pysam problem goes away if we stick with `1.0.4`? Since we are deprecating it, no big reason to upgrade the dep here...

Comment (ctb) [reply setup.py]: it shouldn't, but heck, I'll give it a try ;)

Comment (ctb) [reply setup.py]: there you go - probably cleaner this way (fewer changes), but same failures!

**Review by luizirber — APPROVED** [review]
I think it is too much effort to fix the 10x/bam2fasta usage, for a feature that is optional and is being deprecated. so, LGTM and let's carry on =]

**Pull request merged**